outmon/solvermon.go

```diff
@@ -413,3 +414,14 @@ func (sm *SolverMonitor) reprintFailure(errVertex *vertexMonitor, phaseText stri
 	}
 	errVertex.printError()
 }
+
+
+func (sm *SolverMonitor) modifyId(id string) string {
+	modifiedId := id
+	for old, new := range map[string]string{
+		"transferring docker.io/library":  "transferring locally to docker.io/library",
```

Comment:
this isn't generic, how will this help with other image names?

e.g.

```
./earthly ./examples/python+docker
```

produces

```
              output | [----------] 100% transferring docker.io/earthly/examples:python
```

Reply:
I think there are multiple criterias we have to consider if we want to determine it is transfering locally or remotely. We need to determine if the user's docker is logged in, destination username is the same as the username of the user, then only we can determine if the image would be push locally or remotely @alexcb. Let me know your thoughts. 

Of course, this is assuming that `SAVE IMAGE` only works with docker hub and not other container image repository. If there are other repository type, we may need to identify the different types of syntax

Reply:
I agree -- it should work in a way that's generic to all these use cases.